model_rnn/dataset.py

```diff
@@ -10,24 +10,28 @@
 
 class Dataset(data.Dataset):
     def __init__(self):
-        self.titles = pd.read_csv(DATA_FILEPATHS['item_master'])['PRODUCT_NAME']\
+        self.titles = pd.read_csv(DATA_FILEPATHS['item_master'], encoding='utf-8')['PRODUCT_NAME']\
             .dropna().apply(preprocess_text).astype(str).tolist()
-        self.X, self.y = [], []
-        self.pad_input_size = (0, len(CHAR_TO_IX))
+        self.device = torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')
```

Comment:
Dataset in 99% cases should be CPU-only. Only move data to GPU only before forward and backward pass.